apply-format

```diff
@@ -0,0 +1,13 @@
+#! /bin/bash
+# Apply clang-format using the FL .clang-format on the modified/added cpp/h/hpp files of the current commit:
+clang-format --version || exit 1
+if [ ! -f .clang-format ]; then
+  echo "Cannot find .clang-fornmat in the current dir" 
+  exit 1
+fi
+for file in `git show --name-only -1 --pretty="format:" | grep -E '\.(cpp|h|hpp)$'`; do 
```

Comment:
This is nice - would you mind sending this in a separate PR? There are a few tweaks we'd need to make to this to make sure it touches all of the potential source files :)

Reply:
done